src/SourceLocator/Type/FileIteratorSourceLocator.php

```diff
@@ -25,7 +25,7 @@ class FileIteratorSourceLocator implements SourceLocator
     private $aggregateSourceLocator;
 
     /**
-     * @var \Iterator|\SplFileInfo[]
+     * @var \Iterator<\SplFileInfo>
```

Comment:
@muglug Is there an editor that can understand this? I don't think that eg. PHPStorm supports this syntax.

Reply:
I don't think so? But it's a logical syntax for an iterator that only contains `SplFileInfo`.

The union type that works in PHPStorm - `\Iterator|\SplFileInfo[]` - is *illogical*, since it implies the property can accept arrays.

Reply:
I agree it's logical but this syntax is not supported in editors and other tools (eg. PHPCS).

https://github.com/phpDocumentor/fig-standards/tree/master/proposed is still only a draft.

Reply:
Yup, and until people stop using broken docblocks in their code, there'll be less momentum to fix it.

I have a `@psalm-var` annotation for non-phpdoc-compatible types, I could add it there instead.

Reply:
Yeah, until this can be fixed "properly", at least we have editor support with the `\SplFileInfo[]` hint, which is preferable to not figuring out types IMO :/

Reply:
Syntax change indeed to be reverted here.

Reply:
Yup will do